traitsui/testing/tester/ui_wrapper.py

```diff
@@ -161,12 +161,63 @@ def _perform_or_inspect(self, interaction):
                 with _event_processed():
                     return handler(self, interaction)
 
+        try_DefaultTarget = self._try_interaction_via_DefaultTarget(supported, interaction)
+        if try_DefaultTarget != -1:
+            return try_DefaultTarget
+
         raise InteractionNotSupported(
             target_class=self.target.__class__,
             interaction_class=interaction.__class__,
             supported=supported,
         )
 
+    def _try_interaction_via_DefaultTarget(self, supported, interaction):
+        """ There may not be a handler registered for an interaction on
+        the original target (which is represented by self). Instead, it
+        may be registered for DefaultTarget.  To address this possibility,
+        this method attempts to resolve from the current target to a
+        DefaultTarget, and then sees if there is an appropriate handler
+        that can perform the given interaction on DefaultTarget instead. If
+        so, that handler will ultimately be called and this method returns
+        the result. If not, any interactions for which there are handlers
+        registered for DefaultTarget will be added to the list of supported
+        interactiontypes.
+
+        Parameters
+        ----------
+        supported : List
+            A list of the interactions currently known to be supported for
+            the wrapper object.  If no handler is found for the given
+            interaction with DefaultTarget as the target_class, any handlers
+            supported for DefaultTarget are added to the list. 
+        interaction : instance of interaction type
+            An object defining the interaction.
+
+        Returns
+        -------
+        Any
+            The output of the call to the appropriate handler (if found)
+        None
+            If no appropriate handler is found, the method returns None, and
+            simply adds to the supported list if necessary. 
+        """
+        try:
+            default_target = self.locate(locator.DefaultTarget())
+        # If we can't solve from the current target to DefaultTarget this
+        # interaction isn't supported, and there is nothing to add to supported
+        except LocationNotSupported:
+            pass
+        else:
+            # if we can locate a DefaultTarget try to perform the
+            # interactation on it instead
+            try:
+                return default_target._perform_or_inspect(interaction)
+            # if we can't add to supported
+            except InteractionNotSupported as e:
+                supported.extend(e.supported)
+        return -1
```

Comment:
Returning `-1` for reporting an error is a pattern used in other languages. It has an immediate downside: It means nothing should return `-1` in a normal run condition, so an `inspect(interaction)` that returns `-1` will need to be avoided or worked around somehow. In Python we can raise an exception for it to be captured to avoid this problem.

If this method is to be kept, instead of returning `-1`, we should make sure it raises InteractionNotSupported.
See comment above about reverting the pattern, which means this method will go.

Reply:
Yeah I was worried about this.  The problem I was having is that some of the handlers were returning None (like performing an inspection) and I wanted to check, did this method return a call to a handler, or did it reach completion in which case no handler or solver was found so it adds to supported.  I was struggling to figure out how to separate one form the other.  Just reading within the method makes more sense, but regardless it seems this is likely to get scrapped anyway.